integrationtests.py

```diff
@@ -719,11 +709,22 @@ def _buildAll(self):
             print("Starting compilation of {}".format(sourceFile))
             cxxflags = ["/c", "/nologo", "/EHsc"]
             cmd = CLCACHE_CMD + cxxflags + [sourceFile]
-            processes.append(subprocess.Popen(cmd))
+            processes.append(subprocess.Popen(cmd, env=self._createEnv()))
 
         for p in processes:
             p.wait()
 
+    def _enableNoDirect(self):
+        return False
```

Comment:
I like the idea. But note that this would mean that there is only a single environment for the entire class. In particular, all tests will work on the same cache directory. In any case we should use the constructor to pass the nodirect-parameter.